code/modules/vehicles/scooter.dm

```diff
@@ -0,0 +1,105 @@
+/obj/vehicle/scooter
+	name = "scooter"
+	desc = "A fun way to get around."
+	icon_state = "scooter"
+
+/obj/vehicle/scooter/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/wrench))
+		user << "<span class='notice'>You begin to remove the handlebars...</span>"
+		if(do_after(user, 40/I.toolspeed, target = src))
+			new /obj/vehicle/scooter/skateboard(get_turf(src))
+			new /obj/item/stack/rods(get_turf(src),2)
+			user << "<span class='notice'>You remove the handlebars from [src].</span>"
+			playsound(get_turf(user), 'sound/items/Ratchet.ogg', 50, 1)
+			qdel(src)
+
+/obj/vehicle/scooter/handle_vehicle_layer()
+	if(dir == SOUTH)
+		layer = MOB_LAYER+0.1
+	else
+		layer = OBJ_LAYER
+
+/obj/vehicle/scooter/handle_vehicle_offsets()
+	..()
+	if(buckled_mobs.len)
+		for(var/m in buckled_mobs)
+			var/mob/living/buckled_mob = m
+			switch(buckled_mob.dir)
+				if(NORTH)
+					buckled_mob.pixel_x = 0
+					buckled_mob.pixel_y = 4
+				if(EAST)
+					buckled_mob.pixel_x = -2
+					buckled_mob.pixel_y = 4
+				if(SOUTH)
+					buckled_mob.pixel_x = 2
+					buckled_mob.pixel_y = 4
+				if(WEST)
+					buckled_mob.pixel_x = 0
+					buckled_mob.pixel_y = 4
+
+/obj/vehicle/scooter/skateboard
+	name = "skateboard"
+	desc = "An unfinished scooter which can only barely be called a skateboard. It's still rideable, but probably unsafe. Looks like you'll need to add a few rods to make handlebars."
+	icon_state = "skateboard"
+	vehicle_move_delay = 0//fast
+
+/obj/vehicle/scooter/skateboard/Bump(atom/A)
+	..()
+	if(A.density && buckled_mobs.len)
+		var/mob/living/carbon/human/H = buckled_mobs.len
```

Comment:
@RemieRichards this line is supposed to be defining "H" as the human riding the vehicle (also on second thought I should change that to just living/carbon not living/carbon/human), but apparently "buckled_mobs.len" is not the human riding the vehicle. I need to know what is, because in its current state this just returns as null and creates loadsa runtimes.